File: javatest/src/main/java/com/anymore/javatest/MainActivityViewModel.java
package com.anymore.javatest;

import android.app.Application;
import android.arch.lifecycle.*;
import android.support.annotation.NonNull;

import javax.inject.Inject;

/**
 * Created by liuyuanmao on 2019/2/22.
 */
public class MainActivityViewModel extends AndroidViewModel implements LifecycleObserver {
    private final MutableLiveData<String> mMsg = new MutableLiveData<>();
    @Inject
    public MainActivityViewModel(@NonNull Application application) {
        super(application);
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    public void onStart(){
        mMsg.postValue("onStart");
    }

    public MutableLiveData<String> getMessage() {
        return mMsg;
    }
}

File: kotlindaggertest/src/main/java/com/anymore/kotlindaggertest/MainActivityViewModel.kt
package com.anymore.kotlindaggertest

import android.app.Application
import android.arch.lifecycle.*
import javax.inject.Inject

/**
 * Created by liuyuanmao on 2019/2/22.
 */
class MainActivityViewModel @Inject constructor(application: Application):AndroidViewModel(application),LifecycleObserver{
    private var mMessage = MutableLiveData<String>()


    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    fun onStart(){
        mMessage.postValue("successful!")
    }


    fun getMessage()=mMessage

}